packages/components/src/components/Forms/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  Pressable,
+  StyleProp,
+  View,
+  ViewStyle,
+  useWindowDimensions,
+} from 'react-native'
+import { spacing } from '@department-of-veterans-affairs/mobile-tokens'
+import React, { FC } from 'react'
+
+import { CheckboxRadioProps, FormElementProps } from '../../../types/forms'
+import { ComponentWrapper } from '../../../wrapper'
+import { Description, Error, Header, Hint, Label, fontLabel } from '../FormText'
+import { Icon, IconProps } from '../../Icon/Icon'
+import { Spacer } from '../../Spacer/Spacer'
+import { useTheme } from '../../../utils'
+
+export type CheckboxProps = FormElementProps &
+  CheckboxRadioProps & {
+    /** True to make checkbox appear as checked */
+    checked?: boolean
+    /** True to apply indeterminate icon to checkbox */
+    indeterminate?: boolean
+  }
+
+export const Checkbox: FC<CheckboxProps> = ({
+  checked,
+  label,
+  description,
+  error,
+  header,
+  hint,
+  indeterminate,
+  onPress,
+  required,
+  tile,
+}) => {
+  const theme = useTheme()
+  const fontScale = useWindowDimensions().fontScale
+
+  /**
+   * Container styling
+   */
+  let containerStyle: ViewStyle = {
+    width: '100%',
+  }
+
+  if (error) {
+    containerStyle = {
+      ...containerStyle,
+      borderLeftWidth: spacing.vadsSpace2xs,
+      borderColor: theme.vadsColorFormsBorderError,
+      paddingLeft: spacing.vadsSpaceLg,
+    }
+  }
+
+  /**
+   * Pressable styling
+   */
+  const pressableBaseStyle: StyleProp<ViewStyle> = {
+    width: '100%',
+    flexDirection: 'row',
+    alignItems: 'flex-start',
+  }
+
+  const tileStyle: ViewStyle = {
+    ...pressableBaseStyle,
+    borderWidth: 2,
+    borderRadius: spacing.vadsSpace2xs,
+    padding: spacing.vadsSpaceSm,
+    paddingRight: spacing.vadsSpaceMd,
+    borderColor: checked
+      ? theme.vadsColorFormsBorderActive
+      : theme.vadsColorFormsBorderSubtle,
+    backgroundColor: checked
+      ? theme.vadsColorFormsSurfaceActive
+      : theme.vadsColorSurfaceDefault,
+  }
+
+  /**
+   * Icon
+   */
+  const iconViewStyle: ViewStyle = {
+    // Below keeps icon aligned with first row of text, centered, and scalable
+    alignSelf: 'flex-start',
+    // TODO: Replace lineHeight with typography token
+    minHeight: fontLabel.lineHeight * fontScale,
+    alignItems: 'center',
+    justifyContent: 'center',
+  }
+
+  const iconProps: IconProps = {
+    name: indeterminate
+      ? 'IndeterminateCheckBox'
+      : checked
+        ? 'CheckBox'
+        : 'CheckBoxOutlineBlank',
+    fill:
+      checked || indeterminate
+        ? theme.vadsColorFormsForegroundActive
+        : theme.vadsColorFormsBorderDefault,
+  }
+
+  const _icon = (
+    <View style={iconViewStyle}>
+      <Icon {...iconProps} />
+    </View>
+  )
+
+  return (
+    <ComponentWrapper>
+      <View style={containerStyle}>
+        {header && (
+          <>
+            <Header text={header} />
+            <Spacer size="xs" />
+          </>
+        )}
```

Comment:
Don't love this conditional rendering. Was going back and forth on whether the text components should accept an undefined value for `text` and return `null` in those cases which would clean this up a bit.

Reply:
I'm in favor of moving it into the components and having basically the first line be `if (!text) return null`. 

Seems like a reasonable assumption that they are conditional on the presence of the value, albeit maybe a little less intuitive to track down for fresh eyes.

I am on the fence about then whether you move the Spacer with that or do a line of `{header && hint ? <Spacer /> : null}` between each.

Reply:
Updated FormText elements to handle empty text values and return `null`